pkg/apis/pingcap/v1alpha1/tidbcluster.go

```diff
@@ -498,11 +499,20 @@ func (tc *TidbCluster) PDIsAvailable() bool {
 		}
 	}
 
+	var peerAvailableNum int32
+	for _, pdMember := range tc.Status.PD.PeerMembers {
+		if pdMember.Health {
+			peerAvailableNum++
+		}
+	}
+
+	availableNum += peerAvailableNum
 	if availableNum < lowerLimit {
 		return false
 	}
 
-	if tc.Status.PD.StatefulSet == nil || tc.Status.PD.StatefulSet.ReadyReplicas < lowerLimit {
+	// Response to incluster pod changes. There is a delay for infos in PD.
```

Comment:
Please revise this comment, I do not understand it.

Reply:
I use tc.Status.PD.StatefulSet.ReadyReplicas+peerAvailableNum rather than availableNum to make this part compatible with the origin logic. This comment is the explanation for how I understand the original part.

Reply:
There are some test cases for this comment.